cloudsmith/datadog_checks/cloudsmith/check.py

```diff
@@ -0,0 +1,235 @@
+            ):
+                download_total = response_json['tokens']['downloads']['total']['value']
+            else:
+                self.log.warning("Error when parsing JSON for total token download usage")
+        else:
+            self.log.warning("Error when parsing JSON for tokens")
+
+        entitlement_info = {
+            'token_count': token_count,
+            'token_bandwidth_total': bandwidth_total,
+            'token_download_total': download_total,
+        }
+        return entitlement_info
+
+    def get_parsed_usage_info(self):
+        response_json = self.get_usage_info()
+
+        storage_used = -1
+        bandwidth_used = -1
+        storage_mark = self.UNKNOWN
+        bandwidth_mark = self.UNKNOWN
+
+        if 'usage' in response_json and 'raw' in response_json['usage']:
+            if (
+                'storage' in response_json['usage']['raw']
+                and 'percentage_used' in response_json['usage']['raw']['storage']
+            ):
+                storage_used = response_json['usage']['raw']['storage']['percentage_used']
+                storage_mark = self.OK
+            else:
+                self.log.warning("Error when parsing JSON for storage usage")
+            if (
+                'bandwidth' in response_json['usage']['raw']
+                and 'percentage_used' in response_json['usage']['raw']['bandwidth']
+            ):
+                bandwidth_used = response_json['usage']['raw']['bandwidth']['percentage_used']
+                bandwidth_mark = self.OK
+            else:
+                self.log.warning("Error when parsing JSON for bandwidth usage")
+        else:
+            self.log.warning("Error while parsing JSON for usage information")
+
+        if storage_mark == self.OK:
+            if storage_used >= CRITICAL_QUOTA:
+                storage_mark = self.CRITICAL
+            elif storage_used >= WARNING_QUOTA:
+                storage_mark = self.WARNING
+
+        if bandwidth_mark == self.OK:
+            if bandwidth_used >= CRITICAL_QUOTA:
+                bandwidth_mark = self.CRITICAL
+            elif bandwidth_used >= WARNING_QUOTA:
+                bandwidth_mark = self.WARNING
+
+        usage_info = {
+            'storage_mark': storage_mark,
+            'storage_used': storage_used,
+            'bandwidth_mark': bandwidth_mark,
+            'bandwidth_used': bandwidth_used,
+        }
+        return usage_info
+
+    def check(self, instance):
+        # Use self.instance to read the check configuration
+        if not self.api_key:
+            raise ConfigurationError('Configuration error, please specify api token in conf.yaml.')
+
+        if not self.org:
+            raise ConfigurationError('Configuration error, please specify org in conf.yaml.')
+
+        if not self.base_url:
+            raise ConfigurationError('Configuration error, please specify Cloudsmith url in conf.yaml')
```

Comment:
move this to a `validate_config` method and then add this to `__init__`:
`self.check_initializations.append(self.validate_config)`